### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by saw-jan**
Changes:
- Unit test setup in `jest`
- Added unit test for `first_step` lint rule

Part of https://github.com/gherlint/gherlint-vscode/issues/11

### CODE REVIEW

**Comment by individual-it** [comment]
CC @Talank please review

File: server/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+module.exports = {
+    // All imported modules in your tests should be mocked automatically
+    // automock: false,
+
+    // Stop running tests after `n` failures
+    // bail: 0,
+
+    // The directory where Jest should store its cached dependency information
+    // cacheDirectory: "/tmp/jest_rs",
+
```
Comment (Talank) [inline server/jest.config.js:15]: I think it is better if you do not include unnecessary comments. YAGNI

**Review by kiranparajuli589 — COMMENTED** [review]
overall lgtm :+1:
some points to mention for more improvement of the ci:
- use of caching for npm installation
- v3 can be used  for the setup node part https://github.com/actions/setup-node
- cancel previous running builds for same pr

**Comment by saw-jan** [comment]
> overall lgtm +1 some points to mention for more improvement of the ci:
> 
> * use of caching for npm installation
> * v3 can be used  for the setup node part https://github.com/actions/setup-node
> * cancel previous running builds for same pr

Improvements will be done in further PRs

**Review by Talank — APPROVED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**